src/Compression/Inflater.cs

```diff
@@ -0,0 +1,223 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security;
+
+using ObjCRuntime;
+
+namespace Compression
+{
+	/// <summary>
+	/// Provides a wrapper around the ZLib decompression API.
+	/// </summary>
+	internal sealed class Inflater : IDisposable
```

Comment:
there;'s quite some code duplication in both `Inflater` and `Deflater`, maybe it's with having a base type (or utility class) to share it